gen/fs_home

```diff
@@ -147,16 +153,108 @@ foreach my $login (sort keys %$memberAttributesByLoginAndMount) {
 
 close(SERVICE_FILE);
 
+#----------------------------------------------------------
+# QUOTAS DATA
+#----------------------------------------------------------
+
+#structured data about quotas
+my $dataForUserQuotasByUID = {};
+
+open QUOTAS_FILE,">$quotas_file_name" or die "Cannot open $quotas_file_name: $! \n";
+
+#Temporary define if facility is ready for new quotas (dataQuotas and fileQuotas)
+my $readyForNewQuotas = $facilityAttributes{$A_F_NEW_QUOTAS_READY};
+
+#prepare quotas data
+foreach my $rData (@resourcesData) {
+	my %resourceAttributes = attributesToHash $rData->getAttributes;
+	my @membersData = ($rData->getChildElements)[1]->getChildElements;
+
+	if($readyForNewQuotas) {
+		foreach my $mData (@membersData) {
+			my %memberAttributes = attributesToHash $mData->getAttributes;
+			my $uid = $memberAttributes{$A_UID};
+
+			unless($dataForUserQuotasByUID->{$uid}) {
+				#First process data quotas
+				foreach my $volume (keys %{$memberAttributes{$A_UF_DATA_QUOTAS}}) {
+					my $dataQuota = $memberAttributes{$A_UF_DATA_QUOTAS}{$volume};
+
+					my $softDataQuota = $dataQuota;
+					$softDataQuota =~ s/:.*$//;
+					my $hardDataQuota = $dataQuota;
+					$hardDataQuota =~ s/^.*://;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$DATA_QUOTA_HEADER} = quotaToKb $softDataQuota;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$DATA_LIMIT_HEADER} = quotaToKb $hardDataQuota;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$FILE_QUOTA_HEADER} = 0;
+					$dataForUserQuotasByUID->{$uid}->{$volume}->{$FILE_LIMIT_HEADER} = 0;
```

Comment:
This works, but it deserves at least a comment in code. If a user will be assigned to 2 or more resources, you will first set data quota to correct value and file quota to zero. Then you will set file quota to correct number. After that you will start to process second resource, so you set again data quota to correct number (which is OK), but you also reset file quota to zero. Then you move to next part when you set file quota back to correct number.

Reply:
I don't think so. There is this "if clause:
`unless($dataForUserQuotasByUID->{$uid})`
which will prevent reading of these user_facility attributes more than once per facility.

Reply:
Good point. This section processes only user-facility attributes, therefore it's enough for it to run once for each user.